driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -621,9 +641,27 @@ private static void updateLocationInfo(
     if (!isInitialConnection) cluster.loadBalancingPolicy().onAdd(host);
   }
 
-  private static void refreshNodeListAndTokenMap(
-      Connection connection,
-      Cluster.Manager cluster,
+  /**
+   * Submits query to the appropriate system.peers table based on whether or not {@link
+   * ProtocolOptions#isAllowHostPortDiscovery()} is enabled.
+   *
+   * @param connection connection to send request on.
+   * @return result of peers query.
+   */
+  private ListenableFuture<ResultSet> selectPeersFuture(final Connection connection) {
+    String query =
```

Comment:
This is very explicit now, the `peers_v2` is only ever queried if `allowHostPortDiscovery` is used, and if the query fails, cluster initialization fails.

Reply:
One thing to think about here is whether or not we should fall back to the old system peers query if we get an invalid query request back. I think in that specific case it might be sane to fall back on old peers query.

Reply:
I've wavered back and forth on this, my current line of thinking is if the user opts in using  `allowHostPortDiscovery`, they are intending to depend on the presence of the peers_v2 table to tell them what ports to use.  In absence of that table, I think failing is the right thing to do.  Whereas with java driver 4.x, we always check this table first, so in that case it makes sense to downgrade to the peers table.